### PR TITLE
Add 'safe' check-if-vehicle-is-flying arm/disarm options

### PR DESCRIPTION
**Pull request by peterbarker**
See https://discuss.ardupilot.org/t/switch-disarm-while-in-flight-crashed-how-to-make-disarming-safer/40282/22

### CODE REVIEW

File: Tools/autotest/ArduPlane_Tests/CheckedDisarm/flaps.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+QGC WPL 110
```
Comment (IamPete1) [inline Tools/autotest/ArduPlane_Tests/CheckedDisarm/flaps.txt:1]: is this mission used?

**Comment by IamPete1** [comment]
To better explain a alternate. Instead of a safe disarm how about being allowed to re-arm after a switch disarm with no checks for `DISARM_DELAY` seconds, this gives the user the ability to fix there mistake. Of course it also adds a different set of edge cases for people to hit.

**Comment by rmackay9** [comment]
With power comes responsibility.  If we have an instant disarm auxiliary switch some vehicles are going to be lost due to pilot error.

We had a lengthy discussion about this on the dev call and decided to delay merging this because I'm concerned that it will be difficult for users to pick the correct option for their situation and that situation may change.

As a recap we have these existing options:

- Disarm with the sticks: disarms only when landing detector says vehicle is landed.  Sometimes users complain they cannot disarm however because the landing detector doesn't detect the landing especially after a rough landing
- Disarm with the GCS.  theoretically we have "safe" and "force" options but my understanding is MP at least always sends the "force" option so the vehicle always "force" disarms.  It pops up an "are you sure?" message box to reduce the chance of accidental disarms.
- Arm/Disarm switch: always disarms but users may accidentally move the switch in-flight.  They may not be able to re-arm because the vehicle may have flipped upside down soon after and so arming will fail
- EStop switch: always disarms but the vehicle has 10 seconds or so to re-arm.  This is safer in case the user accidentally moves the switch but can be more dangerous as well because users may not wait the 10 seconds for the disarm after landing and could accidentally move the switch again and the motors could start spinning again

So this PR's proposal is to add an Arm/**Safe** Disarm option.  The problem is that users will complain, "why didn't the vehicle disarm when I commanded a disarm with the switch?" in cases where the landing detector didn't work or the vehicle was still flying.

From a support point of view we will have a reasonable answer for both situations (i.e. where the user wanted to disarm but couldn't and the case where the pilot accidentally disarmed the vehicle) because we can say, "ah, you should have used the other auxiliary switch option" but in practice I'm not sure any more vehicles will be saved because either situation could happen to the same pilot/vehicle on different days.

After further thoughts on this I'm OK to add this auxiliary switch option but I don't expect it will significantly reduce support load or save many vehicles.

**Comment by peterbarker** [comment]
I think I'll close this one - Randy's concerns are legit.

@Hwurzburg do you think we ought to drop a note in the Wiki against the arm/disarm on a switch that users should consider some sort of mechanical or logical interlock on their transmitters for some functions (e.g. arm/disarm on switch)?

**Comment by Hwurzburg** [comment]
@peterbarker there is a warning in the wiki re disarm with switch that it it unconditional...

File: Tools/autotest/ArduPlane_Tests/CheckedDisarm/flaps.txt (hunk 1 above)
Comment (peterbarker) [reply Tools/autotest/ArduPlane_Tests/CheckedDisarm/flaps.txt:1]: Yeah, for fly_home_land_and_disarm I think,.

**Pull request closed without merging**